constraint-solver/src/constraint_system.rs

```diff
@@ -117,20 +129,27 @@ impl<T: FieldElement, V: Clone + Hash + Ord + Eq>
     pub fn solve(
         &self,
         bus_interaction_handler: &dyn BusInteractionHandler<T>,
-        range_constraints: &impl RangeConstraintProvider<T, V>,
+        range_constraint_provider: &impl RangeConstraintProvider<T, V>,
     ) -> Result<Vec<Effect<T, V>>, ViolatesBusRules> {
-        let range_constraints = self.to_range_constraints(range_constraints);
+        let range_constraints = self.to_range_constraints(range_constraint_provider);
         let range_constraints =
             bus_interaction_handler.handle_bus_interaction_checked(range_constraints)?;
         Ok(self
             .fields()
             .zip_eq(range_constraints.fields())
-            .filter_map(|(expr, rc)| {
-                // TODO: Handle range constraint updates for complex expressions.
-                if let Some(var) = expr.try_to_simple_unknown() {
-                    return Some(Effect::RangeConstraint(var, rc.clone()));
-                }
-                None
+            .filter(|(expr, _)| expr.is_affine())
```

Comment:
Could you leave a TODO to handle more complex cases as well? I think we could in the future, by introducing variables for the bus interactions parameters, like we used to.